worlds/overcooked2/__init__.py

```diff
@@ -0,0 +1,509 @@
+import os
+import json
+
+from enum import Enum
+from typing import Callable, Dict, Any, List, Optional
+
+from BaseClasses import ItemClassification, CollectionState, Region, Entrance, Location, RegionType, Tutorial
+from worlds.AutoWorld import World, WebWorld
+
+from .Overcooked2Levels import Overcooked2Level, Overcooked2GenericLevel
+from .Locations import Overcooked2Location, oc2_location_name_to_id, oc2_location_id_to_name
+from .Options import overcooked_options, OC2Options, OC2OnToggle
+from .Items import item_table, is_progression, Overcooked2Item, item_name_to_id, item_id_to_name, item_to_unlock_event, item_frequencies
+from .Logic import has_requirements_for_level_star, has_requirements_for_level_access, level_shuffle_factory, is_item_progression, is_useful
+
+
+class Overcooked2Web(WebWorld):
+    theme = "partyTime"
+
+    bug_report_page = "https://github.com/toasterparty/oc2-modding/issues"
+    setup_en = Tutorial(
+        "Multiworld Setup Tutorial",
+        "A guide to setting up the Overcooked! 2 randomizer on your computer.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["toasterparty"]
+    )
+
+    tutorials = [setup_en]
+
+
+class PrepLevelMode(Enum):
+    original = 0
+    excluded = 1
+    ayce = 2
+
+
+class Overcooked2World(World):
+    """
+    Overcooked! 2 is a franticly paced arcade cooking game where
+    players race against the clock to complete orders for points. Bring
+    peace to the Onion Kingdom once again by recovering lost items and abilities,
+    earning stars to unlock levels, and defeating the unbread horde. Levels are
+    randomized to increase gameplay variety. Play with up to 4 friends.
+    """
+
+    # Autoworld API
+
+    game = "Overcooked! 2"
+    web = Overcooked2Web()
+    option_definitions = overcooked_options
+    topology_present: bool = False
+    remote_items: bool = True
+    remote_start_inventory: bool = False
+    data_version = 1
+    base_id = 0
```

Comment:
as far as i can tell this isn't being used anywhere? you can also remove topology and remote_items if desired since those default to your current values. Since you have entrance shuffle this is brand new but should be useful if you want players to get server hints such as "Fire Extinguisher is at Level 2-3(location) at Level 1-1(entrance)" https://github.com/ArchipelagoMW/Archipelago/commit/c2d69cb05eaaa1514b324d1ef4fbd1ff38444130